cmd/zdb/zdb.c

```diff
@@ -2651,8 +2666,14 @@ visit_indirect(spa_t *spa, const dnode_phys_t *dnp,
 				break;
 			fill += BP_GET_FILL(cbp);
 		}
-		if (!err)
-			ASSERT3U(fill, ==, BP_GET_FILL(bp));
+		if (!err) {
+			if (fill != BP_GET_FILL(bp)) {
+				(void) printf("%16llx: Block pointer "
```

Comment:
print to stderr instead?

Reply:
That's what my first commit did.  But I switched to stdout in response to @amotin's comments.

Reply:
I think printing to stderr makes sense only if we exit immediately, so it is the last message before we return error status.  But if we continue, then using different stream makes it difficult to understand what's actually wrong.  Om the other side, if we print it to stdout, then it has to be formatted in some way nice to possible parsers or readers, unless we expect them to ignore all the output after getting an errors.